src/widget/textbox.rs

```diff
@@ -14,18 +14,28 @@
 
 //! A textbox widget.
 
+use std::cmp::{max, min};
+use std::ops::Range;
+use std::time::{Duration, Instant};
+
 use crate::{
-    Action, BaseState, BoxConstraints, Cursor, Env, Event, EventCtx, KeyCode, LayoutCtx, PaintCtx,
-    UpdateCtx, Widget,
+    Action, BaseState, BoxConstraints, Cursor, Env, Event, EventCtx, HotKey, KeyCode, LayoutCtx,
+    PaintCtx, RawMods, SysMods, TimerToken, UpdateCtx, Widget,
 };
 
 use crate::kurbo::{Affine, Line, Point, RoundedRect, Size, Vec2};
-use crate::piet::{Color, FontBuilder, Piet, RenderContext, Text, TextLayout, TextLayoutBuilder};
+use crate::piet::{
+    Color, FontBuilder, PietText, PietTextLayout, RenderContext, Text, TextLayout,
+    TextLayoutBuilder,
+};
+
+use crate::unicode_segmentation::GraphemeCursor;
 
 const BACKGROUND_GREY_LIGHT: Color = Color::rgba8(0x3a, 0x3a, 0x3a, 0xff);
```

Comment:
one last thought: since env has landed, we can start moving these sorts of things into that, if we like? But I'm happy for this to be followup.

Reply:
I was planning on a styled widgets PR next, so I think I'd prefer to merge now.

Reply:
sounds good!